src/main/java/com/autotune/common/data/system/info/device/ContainerDeviceList.java

```diff
@@ -0,0 +1,130 @@
+package com.autotune.common.data.system.info.device;
+
+import com.autotune.analyzer.utils.AnalyzerConstants;
+import com.autotune.common.data.system.info.device.accelerator.AcceleratorDeviceData;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class ContainerDeviceList implements DeviceHandler, DeviceComponentDetector {
+    private final HashMap<AnalyzerConstants.DeviceType, ArrayList<DeviceDetails>> deviceMap;
+    private boolean isAcceleratorDeviceDetected;
+    private boolean isCPUDeviceDetected;
+    private boolean isMemoryDeviceDetected;
+    private boolean isNetworkDeviceDetected;
+
+    public ContainerDeviceList(){
+        this.deviceMap = new HashMap<AnalyzerConstants.DeviceType, ArrayList<DeviceDetails>>();
+        this.isAcceleratorDeviceDetected = false;
+        // Currently setting up CPU, Memory and Network as true by default
+        this.isCPUDeviceDetected = true;
+        this.isMemoryDeviceDetected = true;
+        this.isNetworkDeviceDetected = true;
+    }
+
+    @Override
+    public void addDevice(AnalyzerConstants.DeviceType deviceType, DeviceDetails deviceInfo) {
+        if (null == deviceType || null == deviceInfo) {
+            // TODO: Handle appropriate returns in future
+            return;
+        }
+
+        if (deviceType == AnalyzerConstants.DeviceType.ACCELERATOR)
+            this.isAcceleratorDeviceDetected = true;
+
+        // TODO: Handle multiple same entries
+        // Currently only first MIG is getting added so no check for existing duplicates is done
+        if (null == deviceMap.get(deviceType)) {
+            ArrayList<DeviceDetails> deviceDetailsList = new ArrayList<DeviceDetails>();
+            deviceDetailsList.add(deviceInfo);
+            this.deviceMap.put(deviceType, deviceDetailsList);
+        } else {
+            this.deviceMap.get(deviceType).add(deviceInfo);
+        }
+    }
+
+    @Override
+    public void removeDevice(AnalyzerConstants.DeviceType deviceType, DeviceDetails deviceInfo) {
+        if (null == deviceType || null == deviceInfo) {
+            // TODO: Handle appropriate returns in future
+            return;
+        }
+        // TODO: Need to be implemented if we need a dynamic experiment device updates
+        if (deviceType == AnalyzerConstants.DeviceType.ACCELERATOR) {
+            if (null == deviceMap.get(deviceType) || this.deviceMap.get(deviceType).isEmpty()) {
+                this.isAcceleratorDeviceDetected = false;
+            }
+        }
+    }
+
+    @Override
+    public void updateDevice(AnalyzerConstants.DeviceType deviceType, DeviceDetails deviceInfo) {
+        // TODO: Need to be implemented if we need a dynamic experiment device updates
+    }
+
+    @Override
+    public DeviceDetails getDeviceByParameter(AnalyzerConstants.DeviceType deviceType, String matchIdentifier, AnalyzerConstants.DeviceParameters deviceParameters) {
```

Comment:
javadoc here for the usecases and document the valid return values

Reply:
Added the javadoc and use case comments